File: Chess/src/main/java/game/boards/Tile.java
package game.boards;

import game.pieces.Coordinate;
import game.pieces.Piece;

public class Tile implements java.io.Serializable {

    private Piece piece;
    private Coordinate coordinate;

    public Tile(Coordinate coordinate, Piece piece) {
        this.setPiece(piece);
        this.setCoordinate(coordinate);
    }

    public Piece getPiece() {

        return this.piece;
    }

    public void setPiece(Piece p) {
        this.piece = p;
    }

    public Coordinate getCoordinate() {

        return this.coordinate;
    }

    public void setCoordinate(Coordinate coordinate) {
        this.coordinate = coordinate;
    }

    public void setCoordinate(int x, int y) {
        this.coordinate.setX(x);
        this.coordinate.setY(y);
    }

    public boolean hasPiece() {
        return this.piece != null;
    }

    @Override
    public String toString() {
        return coordinate.toString() + " Piece " + ((hasPiece() ? piece.toString() : "Empty"));
    }


}

File: Chess/src/main/kotlin/hummel/game/board/Tile.kt
package main.kotlin.hummel.game.board

import main.kotlin.hummel.game.piece.Coordinate
import main.kotlin.hummel.game.piece.Piece
import main.kotlin.hummel.game.piece.PieceNull
import java.io.Serializable

open class Tile(var coordinate: Coordinate, var piece: Piece) : Serializable {
    fun hasPiece(): Boolean {
        return piece !is PieceNull
    }

    override fun toString(): String {
        return coordinate.toString() + " Piece " + if (hasPiece()) piece.toString() else "Empty"
    }
}